chapters/playframework/index.md

```diff
@@ -0,0 +1,546 @@
+The image below [by Julien Richard-Foy](https://www.packtpub.com/books/content/handling-long-running-requests-play) (major contributor of Play) illustrates this: 
+the execution of code is interleaved, while the threaded model would block the entire thread.
+
+![Threaded vs evented execution models](images/reactive/execution-models.png)
+
+Until this point, this section could have been about NodeJS or any other modern platform as well.
+While basic I/O requires only some minor usage of Scala's Futures (or Promises) which can be then handed off to asynchronous actions, 
+for more complex operations Play offers some tools that are unique, some to Scala, but some are really in Play itself.
+
+For example, when we defer a heavy computation, we need to put it into a different execution context, 
+or we would still run out of threads on the default execution context. We can do so using very little code in Play:
+
+In `application.conf`:
+
+{% highlight text %}
+
+my-context {
+  fork-join-executor {
+    parallelism-factor = 20.0
+    parallelism-max = 200
+  }
+}
+
+{% endhighlight %}
+
+In `MyController.scala`:
+
+{% highlight scala %}
+
+implicit val myExecutionContext = Akka.system.dispatchers.lookup("my-context")
+def index = Action.async {
+  val futureInt = scala.concurrent.Future { intensiveComputation() }
+  futureInt.map(i => Ok("Got result: " + i))
+}
+
+{% endhighlight %}
+
+Julien Richard-Foy visualised this in the image below. 
+The two white horizontal boxes represent the execution contexts with both two threads.
+Using execution contexts, related work can be grouped, keeping the default context fast and responsive while still waiting for legacy blocking operations on other threads, in a transparent way.
+
+![Execution Contexts](images/reactive/2400OS_05_08.png)
+
+However, in general, blocking operations should be prevented. 
+It is better to divide the work into small tasks that can be executed individually.
+At the core of Play lives the [very high level](https://john-millikin.com/articles/understanding-iteratees/) [Iteratee library](https://www.playframework.com/documentation/latest/Iteratees), which follows the 
+[Consumer/Producer](http://en.wikipedia.org/wiki/Producer%E2%80%93consumer_problem) pattern. 
+It is used to handle incoming requests as streams of 
+[Byte](http://www.scala-lang.org/api/current/index.html#scala.Byte)'s, 
+which finally get converted to a 
+[Result](https://www.playframework.com/documentation/2.4.x/api/scala/index.html#play.api.mvc.Result), 
+which gets served to the client. 
+The Iteratee library is not limited to HTTP tasks however, it can iterate over and result into all types.
+The library itself consists of three types, the Enumerator, producing data, 
+the Enumeratee which transforms, maps or interleaves data,
+and the Iteratee which consumes data, producing a final result:
+
+````Enumerator (produce data) → Enumeratee (map data) → Iteratee (consume data)````
```

Comment:
The only reason this works in Github is because they allow slightly more

Either use `Enumerator (produce data) → Enumeratee (map data) → Iteratee (consume data)` (single backtick)

or

```
Enumerator (produce data) → Enumeratee (map data) → Iteratee (consume data)
```

which is like

```
'''
Enumerator (produce data) → Enumeratee (map data) → Iteratee (consume data)
'''
```

where a `'` symbolizes a backtick (which i cant type here)

Optionally use

```
'''scala
Enumerator (produce data) → Enumeratee (map data) → Iteratee (consume data)
'''
```

for specific highlighting